src/Servers/IIS/IIS/test/Common.FunctionalTests/HttpsTests.cs

```diff
@@ -87,10 +89,26 @@ public async Task ServerAddressesIncludesBaseAddress()
 
             var deploymentResult = await DeployAsync(deploymentParameters);
             var client = CreateNonValidatingClient(deploymentResult);
-
             Assert.Equal(deploymentParameters.ApplicationBaseUriHint + appName, await client.GetStringAsync($"/{appName}/ServerAddresses"));
         }
 
+        [ConditionalFact]
+        [RequiresNewHandler]
+        public async Task CheckProtocolIsHttp2()
+        {
+            var port = TestPortHelper.GetNextSSLPort();
+            var deploymentParameters = Fixture.GetBaseDeploymentParameters(HostingModel.InProcess);
+            deploymentParameters.ApplicationBaseUriHint = $"https://localhost:{port}/";
+            deploymentParameters.AddHttpsToServerConfig();
+            deploymentParameters.SetWindowsAuth(false);
+
+            var deploymentResult = await DeployAsync(deploymentParameters);
+            var client = CreateNonValidatingClient(deploymentResult);
+            client.DefaultRequestVersion = HttpVersion.Version20;
+
+            Assert.Equal("HTTP/2", await client.GetStringAsync($"/CheckIsHttp2"));
```

Comment:
Let's test other protocol versions too?

Reply:
Other tests already verify this behavior (checking http 1.1)